src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -1,60 +1,177 @@
 package co.inventorsoft.academy.collections.model;
 
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.Set;
+import java.util.*;
+import java.util.function.Function;
 
-public class Range<T> implements Set<T> {
+public class Range<T> implements Set<T> , Comparable {
+
+    static Object[] elementData = new Object[10];
+    static int modCount = 0;
+
+    static Number left;
```

Comment:
You don't have to use **static**. It's better to avoid using static at all (except when you really want to). Also there will be  pitfalls with concurrent execution, you will have one **left**, **right**,... for all Range's.